### PR TITLE
Pylint tests misc

### PR DESCRIPTION
**Pull request by marcospri**
This starts fixing pylint issues in the code under `tests` 

I'll bunch a few fixes together in the same PR. On the `h` package there's still a few fixed missing but there's a lot of code under tests so i'll start sending a few like this one.

### CODE REVIEW

File: tests/functional/api/test_flags.py [hunk 1]
```diff
@@ -5,7 +5,7 @@ class TestPutFlag:
     def test_it_returns_http_204_if_user_allowed_to_flag_shared_annotation(
         self, app, annotation, user_with_token
     ):
-        user, token = user_with_token
+        _, token = user_with_token
```
Comment (jon-betts) [inline tests/functional/api/test_flags.py:8]: This is super weird. Why do we have a fixture where we get two bits but only want one? Seems pretty clear we'd like the token a lot?

File: tests/h/app_test.py [hunk 2]
```diff
@@ -24,7 +24,7 @@ def test_it_configures_pyramid_sentry_plugin(self, pyramid_config):
     def pyramid_config(self, pyramid_config):
         # Mock out jinja2 related stuff
         pyramid_config.get_jinja2_environment = mock.create_autospec(
-            spec=lambda: JinjaEnvironment()
+            spec=lambda: JinjaEnvironment()  # pylint: disable=unnecessary-lambda
```
Comment (jon-betts) [inline tests/h/app_test.py:27]: Is this getting a the idea that the thing is a thing which accepts no arguments and returns itself?

```suggestion
            spec=JinjaEnvironment
```

I think this is _true_, but kind of obscure from a testing point of view. As we're trying to fake a function which returns a `JinjaEnvironment`, and while a `JinjaEnvironment` is a callable that does this, it's not what we really mean?

Something Sean suggested on other tickets is like this:

```suggestion
            spec=pyramid_config.get_jinja2_environment
```

Not sure if that would work here

File: tests/h/search/query_test.py [hunk 3]
```diff
@@ -944,7 +944,7 @@ def test_matches_replies_of_replies_to_an_annotation(self, Annotation, search):
 
 class TestTagsAggregation:
     def test_it_returns_annotation_counts_by_tag(self, Annotation, search):
-        for i in range(2):
+        for _ in range(2):
             Annotation(tags=["tag_a"])
```
Comment (jon-betts) [inline tests/h/search/query_test.py:948]: This looks a bit odd. Is this a factory fixture? If so `create_batch` or something might be better.

(Can you tell I'm only reviewing this from the diff?)

File: tests/h/services/delete_user_test.py [hunk 4]
```diff
@@ -43,7 +43,7 @@ def test_delete_user_removes_groups_if_no_collaborators(
         self, db_session, group_with_two_users, pyramid_request, svc
     ):
         pyramid_request.db = db_session
-        (group, creator, member, creator_ann, member_ann) = group_with_two_users
+        (group, creator, _, _, member_ann) = group_with_two_users
```
Comment (jon-betts) [inline tests/h/services/delete_user_test.py:46]: Another weird looking fixture, but likely not for today to fix.

File: tests/h/services/oauth_validator_test.py [hunk 5]
```diff
@@ -207,7 +207,7 @@ def test_returns_clients_redirect_uri(self, svc, client):
             client.id, OAuthRequest("ftps://example.org")
         )
 
-        assert "ftps://example.org/auth/callback" == actual
+        assert actual == "ftps://example.org/auth/callback"
```
Comment (jon-betts) [inline tests/h/services/oauth_validator_test.py:210]: :+1: I really hate this as it always breaks my brain about what it's supposed to be in the test output

File: tests/h/util/document_claims_test.py [hunk 6]
```diff
@@ -794,5 +794,4 @@ def one(list_):
 def strip_prefix(prefix, s):
     if s.startswith(prefix):
         return s[len(prefix) :]
-    else:
-        return s
+    return s
```
Comment (jon-betts) [inline tests/h/util/document_claims_test.py:797]: Looks like should be hitting the invalid variable name here? Is that disabled for the tests?

**Review by jon-betts — APPROVED** [review]
Looks good. I had a few questions about things, but nothing worth not merging on.

There are some super weird fixtures going on here, but trying to fix any of that is totally out of scope for a linting PR.

File: tests/h/util/document_claims_test.py (hunk 6 above)
Comment (marcospri) [reply tests/h/util/document_claims_test.py:797]: We probably are, but as I go thru this by "rule" I sometimes have to visit the same line more than once.

File: tests/h/app_test.py (hunk 2 above)
Comment (marcospri) [reply tests/h/app_test.py:27]: I tried the first suggestion before sending the PR and didn't work and neither does `spec=pyramid_config.get_jinja2_environment`

**Pull request merged**